### ISSUE TITLE
Manual upload of multiple files hangs

### ISSUE THREAD

**Issue by h-castle** [nextcloud/android#3715]
### Actual behaviour
Manual upload of multiple files hangs in between finishing previous bunch of files and starting next file until:
- the app restarted or
- "uploads" page refreshed or 
- screen rotated.

It's not always enough to just refresh the page, most of the time I need to restart the client, but sometimes it's back to work after rotating screen.

### Expected behaviour
Considering a stable connection to server, no hangs should occur during transfer.
 
### Steps to reproduce
(On client:)
1. create 100 files (~50MB per file)
2. select all files created on step 1 and start uploading them to your server.

### Environment data
General Notes:
All optimizations disabled and all required privileges are granted to the app.

Android version:
MIUI Global 10.1.1.0

Device model:
Redmi 4x, Xiaomi

Stock or customized system:
stock

Nextcloud app version:
3.5.0

Nextcloud server version:
15

**Comment by MeFisto94** [comment]
Can confirm, have this issue as well on a freshly installed nextcloud on docker.
100 Files with roughly 2 MB (Pictures), I can see in the Uploaded Tab how it stops uploading after 3-4 items and I have to re-open the app or e.g. click on the notification (Uploading IMG1234.jpg) or other things for it to continue for another 3-4 images.

From the Access Logs I found out "roughly" where it hangs.
It usually does GET /index.php, HEAD theFile, MKCOL some random filename, PROPFIND that name, PUT that name, MOVE that name and then a PROPFIND on theFile.

When the issue appears, it stops after the very first MKCOL. It doesn't proceed to PROPFIND that name.

Device in this case is a Stock Galaxy S7 edge

**Comment by GoCyberEd** [comment]
I have a similar issue. Only the first few of the files transfer, then it stops. Could be a coincidence, but it always seems to be six files for me. After that, the upload just gives up. No notification/etc. Think this is related? If not I'll make a new issue.

**Comment by h-castle** [comment]
@GoCyberEd after next such stop please try to restart the app and see if the upload is renewed.

**Comment by GoCyberEd** [comment]
I tested this feature again, this time 7 out of 12 of the selected photos were uploaded. The notifications popped up one by one for the first seven. After that, there were no further notifications or any indication that there were files remaining to be uploaded. 

I then killed the app and restarted it. Still nothing. If I visit the uploads section of the app, it only shows the successful uploads (not the missing 5).

I'm using the latest version available on Google Play (3.6.1)

**Comment by h-castle** [comment]
@GoCyberEd did you check weren't these five marked as failed.?
but if upload is really not resumed, it makes sense to open a new ticket. please mention this issue as similar.
Later I will open ticket on server side also.

**Comment by BilBomba** [comment]
I got the same problem.

**Comment by omgmisterbill** [comment]
Can confirm similar issue as well.  15.0.8 with android.
I thought it might be power or sleep settings for android but I believe I've turned everything off fit the NC app.
I've been meaning to configure auto uploads for a while but now that I have,  I essentially have to select the 'uploading' notification every minute or so to keep the upload going.

**Comment by omgmisterbill** [comment]
Ok I tried harder and my issue was 'allow background activity'.  Doing some googling shows maybe it changes back every reboot.   Not sure.

**Comment by piojan** [comment]
I am having a similar problem with manual upload hanging while going out of the app or screen locks.
I has the app protected by the internal security. After disabling it, the whole bunch of files awaiting upload where correctly processed in the background. Are you using application protection with pin or internal (smartphone security)?

**Comment by h-castle** [comment]
@piojan no per app means of security have been used.

**Comment by shootan** [comment]
I have a similar issue.
NextCloud version 16.0.1, and Nextcloud app version 3.7.0.
When uploading multiple photo files, only 3 to 4 files are uploaded and the next upload is not executed.
This happens when the home key is used to bring down the application or when the power button is pressed to turn the screen off.
If you are on the Upload tab while keeping the screen off, it may work normally.
This issue is a big problem for using Nextcloud.

**Comment by MaartenMees** [comment]
Same issue

**Comment by marko-k0** [comment]
I am experiencing the same issue as well.

**Comment by fractalf** [comment]
Same here..

I'm running latest docker image from https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/insecure/mariadb-cron-redis/fpm/docker-compose.yml

Seems to be an Android client (tried from Play Store and F-Droid) problem, as uploading multiple images from the web-client works fine.

This should be easy to reproduce? Boot up the docker-image and upload 10 images in one go

**Comment by FreeSquare** [comment]
Same symptoms on pixel 3 and freshly installed nextcloud 16. 3-4 pictures are uploaded then it stops

**Comment by FreeSquare** [comment]
Update: the upload resumes when I swipe down to refresh the queue then stops again after 3-4 uploads

**Comment by asim-vax** [comment]
Same issue, exactly the same as @FreeSquare
tried to upload 150+ images, doing it 3-4 at a time is painfully slow

**Comment by veelco** [comment]
Can confirm. I am uploading over 400 photos and I have to refresh on the upload page every 5-6 photos from about 2-3 MB each.

Android: 9
NC app: 3.7.2

**Comment by ZaiLynch** [comment]
Same problem.
Android 10
NC app: 3.7.2
NC: 16.04
Background data enabled.
Upload stalls after a couple of files (0% of new file uploaded) and does not resume.
Works fine when I try it via browser on my laptop, so relatively sure it's the app, rather than the NC setup.

**Comment by camfindlay** [comment]
+1 same issue.
App version 3.8.0
NC: 15.0.8

**Comment by picard001** [comment]
I'm also having this exact behavior.

16.0.5
3.8.0
Android 9 (app power saving exclusion active)

**Comment by ovk** [comment]
Same issue. An example: selecting 20 photos to upload. Only 6 actually uploaded and then upload process stops. The issue occurs always, thus making file upload from Android client essentially unusable.

Client: Android 8.1.0.
Server: Nextcloud 17.0 (deployed using official Docker images).

**Comment by ezaquarii** [comment]
@ovk  Can you make an experiment?

1. hook up your phone to power
2. Settings->Display->Sleep-> some reasonably large value
3. Start upload as previously

I have a sneaking suspicion that it has something to do with power management cutting the network access.

**Comment by ovk** [comment]
@ezaquarii, I just tried the same scenario (uploading 20 photos) with phone charging, sleep time set to 30 minutes, data disabled and only WiFi enabled. I also didn't lock the phone until upload process have finished.

Unfortunately, the result was exactly the same and only 6 photos were uploaded.

I will try attaching adb later on to see if I can get any useful logs from the phone.

**Comment by ovk** [comment]
So I did the same experiment while looking at the logs from my web server (Apache), Nextcloud-FPM container and `adb logcat`. Trying to upload 20 photos, only 6 were uploaded again, and for those 6 I can see image file names in logs and in general logs make sense.

However, what is surprising is that for the missing 14 photos there are no traces of them in any logs, including the ones from an Android device. For each of the 6 successfully uploaded photos I can see logs (grep-ing by image file names) in the `adb logcat` but the other image files don't appear in the logs at all.

Since android logs are huge I only post some of the error logs below that I think may be relevant.
If anyone can provide me with further directions I will be happy to continue debugging the issue.

```
10-20 19:35:00.786  3571 29483 E NotificationService: Package enqueue rate is 5.549114. Shedding 0|com.nextcloud.client|2131821464|null|10100. package=com.nextcloud.client
10-20 19:33:58.257  3571  3585 E NotificationService: Suppressing notification from package by user request.
```

```

10-20 19:33:54.800 14426 14426 D ReceiveExternalFilesActivity: onCreate(Bundle) starting
10-20 19:33:54.826 14426 14426 D ReceiveExternalFilesActivity: onStart() starting
10-20 19:33:54.871 14426 14426 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
10-20 19:33:54.871 14426 14426 I zygote64:   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
10-20 19:33:54.871 14426 14426 I zygote64:   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
10-20 19:33:54.871 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
10-20 19:33:54.871 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
10-20 19:33:54.871 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.populateDirectoryList() (ReceiveExternalFilesActivity.java:696)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.onAccountSet(boolean) (ReceiveExternalFilesActivity.java:241)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.BaseActivity.onStart() (BaseActivity.java:197)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.DrawerActivity.onStart() (DrawerActivity.java:1414)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.FileActivity.onStart() (FileActivity.java:207)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.app.Instrumentation.callActivityOnStart(android.app.Activity) (Instrumentation.java:1334)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.app.Activity.performStart() (Activity.java:7029)
10-20 19:33:54.871 14426 14426 I zygote64:   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2741)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
10-20 19:33:54.871 14426 14426 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
10-20 19:33:54.871 14426 14426 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
10-20 19:33:54.871 14426 14426 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
10-20 19:33:54.871 14426 14426 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/lib/arm64, /data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
10-20 19:33:54.871 14426 14426 I zygote64:   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
10-20 19:33:54.871 14426 14426 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
10-20 19:33:54.871 14426 14426 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
10-20 19:33:54.871 14426 14426 I zygote64:   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
10-20 19:33:54.871 14426 14426 I zygote64:   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
10-20 19:33:54.871 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
10-20 19:33:54.872 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
10-20 19:33:54.872 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.populateDirectoryList() (ReceiveExternalFilesActivity.java:696)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.onAccountSet(boolean) (ReceiveExternalFilesActivity.java:241)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.BaseActivity.onStart() (BaseActivity.java:197)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.DrawerActivity.onStart() (DrawerActivity.java:1414)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.FileActivity.onStart() (FileActivity.java:207)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.app.Instrumentation.callActivityOnStart(android.app.Activity) (Instrumentation.java:1334)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.app.Activity.performStart() (Activity.java:7029)
10-20 19:33:54.872 14426 14426 I zygote64:   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2741)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
10-20 19:33:54.872 14426 14426 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
10-20 19:33:54.872 14426 14426 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
10-20 19:33:54.872 14426 14426 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
10-20 19:33:54.872 14426 14426 I zygote64:
10-20 19:33:54.873 14426 14426 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
10-20 19:33:54.873 14426 14426 I zygote64:   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
10-20 19:33:54.873 14426 14426 I zygote64:   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
10-20 19:33:54.873 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
10-20 19:33:54.873 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
10-20 19:33:54.873 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
10-20 19:33:54.873 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.populateDirectoryList() (ReceiveExternalFilesActivity.java:696)
10-20 19:33:54.873 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.onAccountSet(boolean) (ReceiveExternalFilesActivity.java:241)
10-20 19:33:54.873 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.BaseActivity.onStart() (BaseActivity.java:197)
10-20 19:33:54.873 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.DrawerActivity.onStart() (DrawerActivity.java:1414)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.FileActivity.onStart() (FileActivity.java:207)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.Instrumentation.callActivityOnStart(android.app.Activity) (Instrumentation.java:1334)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.Activity.performStart() (Activity.java:7029)
10-20 19:33:54.874 14426 14426 I zygote64:   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2741)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
10-20 19:33:54.874 14426 14426 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
10-20 19:33:54.874 14426 14426 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/lib/arm64, /data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
10-20 19:33:54.874 14426 14426 I zygote64:   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
10-20 19:33:54.874 14426 14426 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
10-20 19:33:54.874 14426 14426 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
10-20 19:33:54.874 14426 14426 I zygote64:   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
10-20 19:33:54.874 14426 14426 I zygote64:   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
10-20 19:33:54.874 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
10-20 19:33:54.874 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
10-20 19:33:54.874 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.populateDirectoryList() (ReceiveExternalFilesActivity.java:696)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.onAccountSet(boolean) (ReceiveExternalFilesActivity.java:241)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.BaseActivity.onStart() (BaseActivity.java:197)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.DrawerActivity.onStart() (DrawerActivity.java:1414)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.FileActivity.onStart() (FileActivity.java:207)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.Instrumentation.callActivityOnStart(android.app.Activity) (Instrumentation.java:1334)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.Activity.performStart() (Activity.java:7029)
10-20 19:33:54.874 14426 14426 I zygote64:   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2741)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
10-20 19:33:54.874 14426 14426 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
10-20 19:33:54.874 14426 14426 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
10-20 19:33:54.874 14426 14426 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
10-20 19:33:54.874 14426 14426 I zygote64:
10-20 19:33:54.874 14426 14426 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
10-20 19:33:54.874 14426 14426 I zygote64:   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
10-20 19:33:54.874 14426 14426 I zygote64:   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
10-20 19:33:54.874 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
10-20 19:33:54.875 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
10-20 19:33:54.875 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.populateDirectoryList() (ReceiveExternalFilesActivity.java:696)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.onAccountSet(boolean) (ReceiveExternalFilesActivity.java:241)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.BaseActivity.onStart() (BaseActivity.java:197)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.DrawerActivity.onStart() (DrawerActivity.java:1414)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.FileActivity.onStart() (FileActivity.java:207)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.Instrumentation.callActivityOnStart(android.app.Activity) (Instrumentation.java:1334)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.Activity.performStart() (Activity.java:7029)
10-20 19:33:54.875 14426 14426 I zygote64:   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2741)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
10-20 19:33:54.875 14426 14426 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
10-20 19:33:54.875 14426 14426 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/lib/arm64, /data/app/com.nextcloud.client-Mv86T7yETGrQJlNrZzycbg==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
10-20 19:33:54.875 14426 14426 I zygote64:   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
10-20 19:33:54.875 14426 14426 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
10-20 19:33:54.875 14426 14426 I zygote64:   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
10-20 19:33:54.875 14426 14426 I zygote64:   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
10-20 19:33:54.875 14426 14426 I zygote64:   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
10-20 19:33:54.875 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
10-20 19:33:54.875 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
10-20 19:33:54.875 14426 14426 I zygote64:   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.populateDirectoryList() (ReceiveExternalFilesActivity.java:696)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.ReceiveExternalFilesActivity.onAccountSet(boolean) (ReceiveExternalFilesActivity.java:241)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.BaseActivity.onStart() (BaseActivity.java:197)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.DrawerActivity.onStart() (DrawerActivity.java:1414)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.owncloud.android.ui.activity.FileActivity.onStart() (FileActivity.java:207)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.Instrumentation.callActivityOnStart(android.app.Activity) (Instrumentation.java:1334)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.Activity.performStart() (Activity.java:7029)
10-20 19:33:54.875 14426 14426 I zygote64:   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2741)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.os.Looper.loop() (Looper.java:164)
10-20 19:33:54.875 14426 14426 I zygote64:   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
10-20 19:33:54.875 14426 14426 I zygote64:   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
10-20 19:33:54.875 14426 14426 I zygote64:   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
```

```
10-20 19:33:58.357  3571 11298 W ActivityManager: No permission grants found for com.google.android.apps.photos
10-20 19:33:58.401 14278 14289 E DatabaseUtils: Writing exception to parcel
10-20 19:33:58.401 14278 14289 E DatabaseUtils: java.lang.SecurityException: Permission Denial: reading com.google.android.apps.photos.contentprovider.impl.MediaContentProvider uri content://com.google.android.apps.photos.contentprovider/-1/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F23881/REQUIRE_ORIGINAL/NONE/1213385458 from pid=14426, uid=10100 requires the provider be exported, or grantUriPermission()
10-20 19:33:58.401 14278 14289 E DatabaseUtils: 	at android.content.ContentProvider.enforceReadPermissionInner(ContentProvider.java:633)
10-20 19:33:58.401 14278 14289 E DatabaseUtils: 	at android.content.ContentProvider$Transport.enforceReadPermission(ContentProvider.java:503)
10-20 19:33:58.401 14278 14289 E DatabaseUtils: 	at android.content.ContentProvider$Transport.enforceFilePermission(ContentProvider.java:494)
10-20 19:33:58.401 14278 14289 E DatabaseUtils: 	at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:422)
10-20 19:33:58.401 14278 14289 E DatabaseUtils: 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:302)
10-20 19:33:58.401 14278 14289 E DatabaseUtils: 	at android.os.Binder.execTransact(Binder.java:697)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: Not enough permissions to read source file content://com.google.android.apps.photos.contentprovider/-1/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F23881/REQUIRE_ORIGINAL/NONE/1213385458
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: java.lang.SecurityException: Permission Denial: reading com.google.android.apps.photos.contentprovider.impl.MediaContentProvider uri content://com.google.android.apps.photos.contentprovider/-1/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F23881/REQUIRE_ORIGINAL/NONE/1213385458 from pid=14426, uid=10100 requires the provider be exported, or grantUriPermission()
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.os.Parcel.readException(Parcel.java:2013)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:698)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1410)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1247)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.content.ContentResolver.openInputStream(ContentResolver.java:967)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at com.owncloud.android.ui.asynctasks.CopyAndUploadContentUrisTask.doInBackground(CopyAndUploadContentUrisTask.java:147)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at com.owncloud.android.ui.asynctasks.CopyAndUploadContentUrisTask.doInBackground(CopyAndUploadContentUrisTask.java:47)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.os.AsyncTask$2.call(AsyncTask.java:333)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
10-20 19:33:58.408 14426 14450 E CopyAndUploadContentUrisTask: 	at java.lang.Thread.run(Thread.java:764)
```

**Comment by ovk** [comment]
As far as I can tell, the problem is still there.

**Comment by Dolo3000** [comment]
Hi all,

have the same problem with multiple photos (>300) and stopping upload after a few files.

Nextcloud Version: 17.0.1
Android Nextcloud App Version: 3.9.0
Android Version: 9.0

Could solve this by resseting the **"Background data"**-Setting for the Nextcloud app. Problem seems to be Android related. 

On your **Android Phone:**

1. Open Settings
2. Choose "Apps & Notification" 
3. Click on "Nextcloud" App
4. Click on "Advanced"
5. Next choose "Battery"

There should be an option **"Restrict backgroud data"**

In my case it was set to "disable" or "unrestricted".
When the Nextcloud App stops uploading i did the following:

- So i just click on this option and "enabled" the restriction for background data.
- Then set it back to "disabled"

After this the App could upload the files without problem.

Hope this helps!

**Comment by picard001** [comment]
> Hi all,
> 
> have the same problem with multiple photos (>300) and stopping upload after a few files.
> 
> Nextcloud Version: 17.0.1
> Android Nextcloud App Version: 3.9.0
> Android Version: 9.0
> 
> Could solve this by resetting the **"Background data"**-Setting for the Nextcloud app. Problem seems to be Android related.
> 
> On your **Android Phone:**
> 
> 1. Open Settings
> 2. Choose "Apps & Notification"
> 3. Click on "Nextcloud" App
> 4. Click on "Advanced"
> 5. Next choose "Battery"
> 
> There should be an option **"Restrict backgroud data"**
> 
> In my case it was set to "disable" or "unrestricted".
> When the Nextcloud App stops uploading i did the following:
> 
> * So i just click on this option and "enabled" the restriction for background data.
> * Then set it back to "disabled"
> 
> After this the App could upload the files without problem.
> 
> Hope this helps!
> 
> (Edit: grammer stuff..)

I'm on Android 9 (LG G7) and that setting is under Apps & notifications -> Special access -> Unrestricted data access, which provides a list of apps and a toggle.  Mine was set to restricted like most apps.  I don't believe this should affect the app since I'm on WiFi and I have no usage restrictions.

**Comment by GoCyberEd** [comment]
I'd like to point out, even if this is Android related, there must be a work around / different approach the app can (should) use. Other applications (like Amazon photos) automatically upload without a problem even with "restrict background data" in place.

**Comment by iGamerHD** [comment]
For me using a Xiaomi Mi A2 Lite the upload also doesn't work.
I tried to upload about 30 files to my Nextcloud, but the upload process stopped after 4 pics.

I activated the unlimited data access and backgroud data usage before and disabled any battery optimization for the Nextcloud Android app.
So to me it looks like the error is on the Nextcloud app side and not related to Android.
As already mentioned other apps are also able to upload images in the background, even when the device is turned off.

If you need further information from me, just tell me :)

**Comment by xkeysorg** [comment]
Same issue. Sounds like #4633 .

Android 10 
Version 3.9.0

Resetting **Background data** only worked for a few images.

**Comment by Saphirim** [comment]
Problem still exists.
OnePlus 6, Android 10, Nextcloud App-Version 3.9.2

I discovered that a device reboot (after allowing background data and completely disabling any battery optimizations for the app) fixes the issue for several hours/uploads.

But after a night of uploading ~1.5k files I was back at the already described behavior and had to restart the smartphone.

**Comment by Th0maz** [comment]
Same issue here with a Galaxy Tab (Android 9) and Galaxy S10 (Android 10) with latest app version.
However an iOS device is not facing this issue, so I guess it's not a server but android related issue.

**Comment by Th0maz** [comment]
It seems Android Doze might be responsible for this.
Can anyone try and disable battery optimization for the Nextcloud app to see if this works.
It's somewhere under Settings - Apps - Nextcloud - Battery - Battery Optimization - Select 'All' - Nextcloud - Don't Optimize
I tried with uploading a couple of picture files and this setting looks promising.

**Comment by miend** [comment]
I also had this issue, exactly as described by users above, and I painfully managed the upload of over 3000 images from an Android (Oneplus 5 + OxygenOS) device. When interacting with the interface, the uploads would resume for a few files at a time.

Interestingly, once the amount to be uploaded reached somewhere just south of 300 files remaining, all of them completed without issue. So there may be something specifically to due with the quantity in the upload queue.

**Comment by monte-monte** [comment]
Same here. I thought actually it was resolved long time ago, but had just to upload 50+ videos from my Android and had to refresh upload page in the app every 2-3 files uploaded. Hope this will be fixed soon.

**Comment by karnizas85** [comment]
Same here, Pocophone F1 Android 9,Battery optimization disabled.

**Comment by pasdVn** [comment]
Same problem for me (Lineage OS 16.0, galaxy s5, current stable version via f-droid).

I also had a look at adb and the access log of my webserver (nginx) in parallel: What I see is that app always hangs at PROPFIND (after MKCOL, see comment https://github.com/nextcloud/android/issues/3715#issuecomment-470711048) while the last request the web server received was the MKCOL (during the "hanging").
So maybe the PROPFIND is (whyever) not completely transmitted to the the server or something similiar?

**Comment by boggle247** [comment]
> > Hi all,
> > have the same problem with multiple photos (>300) and stopping upload after a few files.
> > Nextcloud Version: 17.0.1
> > Android Nextcloud App Version: 3.9.0
> > Android Version: 9.0
> > Could solve this by resetting the **"Background data"**-Setting for the Nextcloud app. Problem seems to be Android related.
> > On your **Android Phone:**
> > 
> > 1. Open Settings
> > 2. Choose "Apps & Notification"
> > 3. Click on "Nextcloud" App
> > 4. Click on "Advanced"
> > 5. Next choose "Battery"
> > 
> > There should be an option **"Restrict backgroud data"**
> > In my case it was set to "disable" or "unrestricted".
> > When the Nextcloud App stops uploading i did the following:
> > 
> > * So i just click on this option and "enabled" the restriction for background data.
> > * Then set it back to "disabled"
> > 
> > After this the App could upload the files without problem.
> > Hope this helps!
> > (Edit: grammer stuff..)
> 
> I'm on Android 9 (LG G7) and that setting is under Apps & notifications -> Special access -> Unrestricted data access, which provides a list of apps and a toggle. Mine was set to restricted like most apps. I don't believe this should affect the app since I'm on WiFi and I have no usage restrictions.

This did not solve the issue for me. 

Pixel 1, LineageOS 16 with gapps installed, ver 3.10.1 from fdroid. Although this issue has been present ever since I've used nextcloud for the last 2 years, and also on my galaxy S5 phones I had, running Android 7 and 8.

So if the answer is this is androids fault, then it's clearly not going away. So the question is, can we get this fixed in this app or will it's primary function (when used on a phone) remain nearly useless?

I'll contribute whatever is necessary to aid in troubleshooting this issue. Thanks.

**Comment by major-mayer** [comment]
@boggle247 yeah I also noticed something similar. If you do something like swiping, scrolling and continuously restarting the upload queue or even the whole app, eventually the whole queue will be uploaded.
But it takes years and it is a very annoying task if you are trying to upload 100 photos. 
I'm a bit dissapointed that this bug exists for such a long time and receives such few attention, even though it is confirmed by many users and touches such a crucial feature as uploading files.

**Comment by nicolae788** [comment]
Hello. Try to disable notifications among other settings described here. I am running on Android 7.0 and Nextcloud 3.10.1

**Comment by major-mayer** [comment]
@nicolae788 Nah i'm sorry but disabling the notifications of the nextcloud app doesn't change anything for me.
If you stay in the app, the upload process stops after about 5 files/ pictures, if you are outside (display on or off doesn't really make a difference) the upload queue will be also stopped after some time.

**Comment by smacz42** [comment]
Not here for much more than to add my +1. I'd been running a bastardized version of lineageOS on my Moto E2 with this issue, and just chalked it up to so quirk on what I had. I switched to a OnePlus5 and had the same thing happen. I tried un-optimizing the battery usage, and while that seemed to upload _more_ files at once, it still took one of the refreshes noted above to resume the upload, as it appeared to pause very frequently.

I haven't checked the server logs, but have a robust Dev environment and would be willing to dive into them if I could be told what to look for.

**Comment by aardquark** [comment]
I am having the same problem described here, that, unless I continually keep my phone awake by poking and prodding it, the uploads from the Nextcloud app just stop. What I would like to add is that, on the same phone, I was able to use the FolderSync app to upload the same (~675) files to Nextcloud over the webdav interface in less than 30 minutes. I'm thinking that this points to a problem in the Nextcloud app itself. (I don't really want to use FolderSync because it dumps all the photos in a single folder, not using subfolders.)

**Comment by buddebej** [comment]
I experience the same problem. I can confirm that enabling Unrestricted data access for the app fixes the problem.

**Comment by boggle247** [comment]
Alright I enabled it. Although I probably won't be uploading a ton of files for a while. If I do I will also comment whether it helped

**Comment by major-mayer** [comment]
I tried it once again and made sure to disable battery optimization and enable unlimited data access for the Nextcloud app.
My Device is a Xiaomi Mi 8 using an AOSP Rom (so no Xiaomi should be stuff blocking the background processes) which is based on Android 10.
I kept the screen on the whole time and didn't lock the phone or something, but the upload process got stuck anyway.

Here a the logs from the upload procedure:
https://paste.ubuntu.com/p/8QBkHHXCsR/

What I noticed is, that when you open the Nextcloud app and go to the upload queue and refresh, the upload will resume, only to stop again 20 seconds later.
But you can keep reopening the Nextcloud app again and again eventually the upload will finish, but it is very time consuming.

**Comment by tobiasKaminsky** [comment]
@major-mayer thanks for the logs, but I do not see any failing upload.
Can you redo this and state which file is not working?

**Comment by FrederikP** [comment]
@tobiasKaminsky I don't think the issues are failing uploads. It is that the process gets stuck and there is not necessarily any relevant log entry for that. I experienced the same behavior as detailed by @major-mayer and many others in this thread.

**Comment by major-mayer** [comment]
@tobiasKaminsky I also would say that in most cases the upload doesn't actually fail, but just gets stuck and takes forever to process.
I took a small screenrecord to describe the problem that I and many others here experience.

https://www.youtube.com/watch?v=fxA9X8o553Y

As you can see after I reopen the app, the upload resumes, so I don't think that there will be any actual error in the logs.

**Comment by nicolae788** [comment]
Yes, but try to disable notifications. On my phone and os it's doing a bit
better. But i choosed to upload a large quantity of files from the PC...

Alex.

**Comment by nicolae788** [comment]
Did nextcloud updated the android app? I am on 3.11.0 and it seems to be
working correct now.

**Comment by major-mayer** [comment]
Lucky you, for me the situation didn't really change :`(

**Comment by nicolae788** [comment]
But did the version changed? What version do you have? It uploaded a bunch
of files, about 50 with no problems.

**Comment by MaartenMees** [comment]
Hi, I updated the app but still having the same problem. Tried uploading 100 photos a couple of times. None of the times all pictures were uploaded successfully.

**Comment by AaronADev** [comment]
Curious, have there been any updates to this. It's not just photos that this happens with but videos as well.

**Comment by boggle247** [comment]
> Curious, have there been any updates to this. It's not just photos that this happens with but videos as well.

Nope. And this has been an issue far before this issue was initiated. I remember it back in 2017

**Comment by michaeldegroot** [comment]
Just stayed in the uploads tab and nothing is uploading or getting stuck after 5 or 6 pictures. Scrolling, refreshing or changing to diff app and back seems to temporarily fix it.

**Comment by major-mayer** [comment]
Guys test the latest (beta) version, for me it works from 3.12.0 on.
They changed something with the connection handling and I was able to upload 20 photos and videos successfully :) 🎉

**Comment by ezaquarii** [comment]
Fixed in https://github.com/nextcloud/android-library/pull/424

**Comment by ezaquarii** [comment]
Fixed in https://github.com/nextcloud/android-library/pull/424
